components/org.wso2.carbon.security.caas/src/main/java/org/wso2/carbon/security/caas/user/core/claim/InMemoryClaimManager.java

```
@@ -16,23 +16,126 @@
+
+        Map<String, IdentityStoreConnector> identityStoreConnectors = domain.getIdentityStoreConnectorMap();
+
+
+        for (Map.Entry<String, IdentityStoreConnector> identityStoreConnectorEntry :
+                identityStoreConnectors.entrySet()) {
+
+            String connectorId = identityStoreConnectorEntry.getKey();
+            List<MetaClaimMapping> metaClaimMappings = claimMappings.get(connectorId);
+
+            // Create <AttributeName, MetaClaim> map
+            Map<String, MetaClaim> attributeMapping = metaClaimMappings.stream()
+                    .collect(Collectors.toMap(MetaClaimMapping::getAttributeName, MetaClaimMapping::getMetaClaim));
+
+            IdentityStoreConnector identityStoreConnector = identityStoreConnectorEntry.getValue();
+
+            try {
+                String connectorUserId = userManager.getConnectorUserId(user.getUserId(), connectorId);
+
+                List<Attribute> attributeValues = identityStoreConnector.getUserAttributeValues(connectorUserId,
+                        new ArrayList<>(attributeMapping.keySet()));
+                claims.addAll(buildClaims(attributeValues, attributeMapping));
+            } catch (IdentityStoreException | UserManagerException e) {
+                throw new ClaimManagerException("Error retrieving claims for user : " + user.getUserId(), e);
+            }
+        }
+
+        return claims;
     }
 
     @Override
     public List<Claim> getClaims(User user, List<String> claimURIs) throws ClaimManagerException {
-        return null;
+        List<Claim> claims = new ArrayList<>();
+        Domain domain = user.getDomain();
+
+        Map<String, List<MetaClaimMapping>> claimMappings = domain.getClaimMappings();
+
+        Map<String, IdentityStoreConnector> identityStoreConnectors = domain.getIdentityStoreConnectorMap();
+
+
+        for (Map.Entry<String, IdentityStoreConnector> identityStoreConnectorEntry :
```

The same as above I need to throw an Exception in the inner loop.